File: Napakalaki/lib/treasure.rb
# encoding: utf-8

module NapakalakiGame
 
  # class Treasure: representa un tesoro
  class Treasure
    attr_reader :name   # nombre del tesoro
    attr_reader :bonus  # bonus del tesoro
    attr_reader :type   # tipo del tesoro

    def initialize(n, bonus, t)
      @name = n
      @bonus = bonus
      @type = t
    end
    
    def to_s
      "Nombre: #{@name} \nBonus: #{@bonus} \nTipo: #{@type}"
    end

  end # Treasure

end # NapakalakiGame

File: Napakalaki/src/NapakalakiGame/Treasure.java
/**
 * Treasure.java
 */

package NapakalakiGame;

/**
 * Representa un tesoro.
 */
public class Treasure {
    
    private String name;
    private int bonus;
    private TreasureKind type;
    
    public Treasure(String n, int bonus, TreasureKind t) {
        name = n;
        this.bonus = bonus;
        type = t;
    }

    public String getName() {
        return name;
    }

    public int getBonus() {
        return bonus;
    }

    public TreasureKind getType() {
        return type;
    }
    
    @Override
    public String toString() {
        return "Name: " + name + "\nBonus: " + Integer.toString(bonus) 
               + "\nType: " + type;
    }
    
}
